app/TransactionRules/Expressions/ActionExpression.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+/**
+ * ActionExpression.php
+ * Copyright (c) 2024 Michael Thomas
+ *
+ * This file is part of Firefly III (https://github.com/firefly-iii).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+declare(strict_types=1);
+
+namespace FireflyIII\TransactionRules\Expressions;
+
+use Symfony\Component\ExpressionLanguage\ExpressionLanguage;
+use Symfony\Component\ExpressionLanguage\SyntaxError;
+
+class ActionExpression
+{
+    private static array $NAMES = array(
```

Comment:
Fields that can be safely dropped are:

- deleted_at, encrypted

What's the deal with all the `*_count` fields?  Who populates them?

Reply:
I grabbed these from the `TransactionJournal` class, but turns out they were totally wrong. I've updated them to reflect the actual fields present on the `$journal` array.